### PR TITLE
✨Comp backend: allow setting workers idle time before termination (⚠️ devops)

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
- adds new ENV variable on clusters-keeper: ```WORKERS_EC2_INSTANCES_TIME_BEFORE_TERMINATION``` defaults to 3 minutes

## DevOps Checklist
- new variable ```WORKERS_EC2_INSTANCES_TIME_BEFORE_TERMINATION=00:03:00```
- only necessary on AWS-based deployments

### CODE REVIEW

**Review by YuryHrytsuk — APPROVED** [review]

**Review by mrnicegyu11 — APPROVED** [review]
thanks!

File: services/clusters-keeper/src/simcore_service_clusters_keeper/core/settings.py [hunk 1]
```diff
@@ -67,6 +67,11 @@ class WorkersEC2InstancesSettings(BaseCustomSettings):
         "this is required to start a new EC2 instance",
     )
 
+    WORKERS_EC2_INSTANCES_TIME_BEFORE_TERMINATION: datetime.timedelta = Field(
+        default=datetime.timedelta(minutes=3),
```
Comment (pcrespov) [inline services/clusters-keeper/src/simcore_service_clusters_keeper/core/settings.py:71]: MINOR: i know there are different formats that the pydantic parser can consume to produce `timedelta`. Can you add an example of value somewhere. E.g. in the example parameter

File: services/autoscaling/src/simcore_service_autoscaling/core/settings.py [hunk 2]
```diff
@@ -50,23 +50,9 @@ class EC2InstancesSettings(BaseCustomSettings):
...
-    )
-    EC2_INSTANCES_SUBNET_ID: str = Field(
-        ...,
-        min_length=1,
-        description="A subnet is a range of IP addresses in your VPC "
-        " (https://docs.aws.amazon.com/vpc/latest/userguide/configure-subnets.html), "
-        "this is required to start a new EC2 instance",
+    EC2_INSTANCES_CUSTOM_BOOT_SCRIPTS: list[str] = Field(
+        default_factory=list,
+        description="script(s) to run on EC2 instance startup (be careful!), each entry is run one after the other using '&&' operator",
```
Comment (pcrespov) [inline services/autoscaling/src/simcore_service_autoscaling/core/settings.py:55]: THOUGHT: why not add a `validator` that produces a warning are simply stops the app startup (i.e. that is the time this is resolved)?

**Review by pcrespov — APPROVED** [review]
thx. Some thoughts

File: services/autoscaling/src/simcore_service_autoscaling/core/settings.py (hunk 2 above)
Comment (sanderegg) [reply services/autoscaling/src/simcore_service_autoscaling/core/settings.py:55]: what kind of validator? you want me to lint the bash scripts there? like using shellcheck? if yes then it's a bit heavy I think...
if not please precise..

File: services/clusters-keeper/src/simcore_service_clusters_keeper/core/settings.py (hunk 1 above)
Comment (sanderegg) [reply services/clusters-keeper/src/simcore_service_clusters_keeper/core/settings.py:71]: fixed all the settings that are using timedelta, and puts a reference to the pydantic page for description.

**Review by GitHK — APPROVED** [review]

**Pull request merged**